vcd/catalogitem.go

```diff
@@ -69,7 +78,49 @@ func findCatalogItem(d *schema.ResourceData, vcdClient *VCDClient, origin string
 		identifier = d.Get("name").(string)
 	}
 
-	catalogItem, err := catalog.GetCatalogItemByNameOrId(identifier, false)
+	var catalogItem *govcd.CatalogItem
+	if origin == "datasource" {
```

Comment:
For me simplier is 
```
if origin == "datasource" && hasfilter {
...
```

Reply:
hasFilter depends on the "filter" field, which does not exist for resource. Moreover, if I test for both datasource and filter I leave out actions that should happen for datasource without filter.